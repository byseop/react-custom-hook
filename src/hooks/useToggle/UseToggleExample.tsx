import React from 'react';
import useToggle from './useToggle';

export default function UseToggleExmaple() {
  const [toggleState, toggle] = useToggle(false);

  return (
    <>
      <div>{toggleState ? 'true' : 'false'}</div>
      <button onClick={toggle}>toggle button</button>
    </>
  )
}