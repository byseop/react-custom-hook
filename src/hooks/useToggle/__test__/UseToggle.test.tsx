import React from 'react';
import UseToggleExample from '../UseToggleExample';
import { render, fireEvent } from '@testing-library/react';

describe('<UseToggleExample />', () => {
  it('matches snapshot', () => {
    const utils = render(<UseToggleExample />);
    expect(utils.container).toMatchSnapshot();
  });

  it('shows the props correct', () => {
    const utils = render(<UseToggleExample />);
    utils.getByText('false');
    utils.getByText('toggle button');
  });

  it('click button', () => {
    const utils = render(<UseToggleExample />);
    const toggleElement = utils.getByText('false');
    const toggleButton = utils.getByText('toggle button');

    // test event false -> true
    fireEvent.click(toggleButton);

    expect(toggleElement.textContent).toBe('true');
    expect(toggleElement).toHaveTextContent('true');

    // test event true -> false
    fireEvent.click(toggleButton);

    expect(toggleElement.textContent).toBe('false');
    expect(toggleElement).toHaveTextContent('false');
  });
});
