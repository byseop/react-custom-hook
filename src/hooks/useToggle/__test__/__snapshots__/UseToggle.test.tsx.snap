// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseToggleExample /> matches snapshot 1`] = `
<div>
  <div>
    false
  </div>
  <button>
    toggle button
  </button>
</div>
`;
